package/gluon-mesh-vpn-tunneldigger/luasrc/usr/bin/tunneldigger-watchdog

```diff
@@ -33,7 +33,7 @@ local function has_mesh_vpn_neighbours()
 end
 
 if uci:get_bool('tunneldigger', 'mesh_vpn', 'enabled') then
-	if io.popen('pgrep tunneldigger'):read('*l') ~= read_pid_file() then
+	if io.popen('pgrep -x /usr/bin/tunneldigger'):read('*l') ~= read_pid_file() then
```

Comment:
I don't think this will ever match - without `-f`, `pgrep` only matches the process name, never its path (and whether the path is matches with `-f` depends on the way the binary was called).

Reply:
Two options:
- Use `pidof` instead of `pgrep`, which accepts the whole path (I have only checked a full-featured pidof though, not busybox)
- Check whether `/proc/$PID/exe` points to the right binary - which is also more efficient than using pgrep or pidof

Reply:
I can `pgrep -x /usr/bin/tunneldigger` on two nodes and it finds 3 PIDs on each; the same it finds with `pgrep tunneldigger`.

Reply:
I also adjusted the watchdog like here on one of the nodes and confirmed that it does not just restart tunneldigger.

So, to me it looks like this works as intended?

Reply:
> _Edit_: This only matches when the binary was called using the whole path.

I checked it out on several nodes with different hardware and tunneldigger was always started by using full path. In case of further trouble, I´d suggest to rename the watchdog to something different, e.g. tunneldigger-watchdog -> td-watchdog
This would help to avoid such grep-confusion, too